text/src/grep.rs

```diff
@@ -0,0 +1,448 @@
+//
+// Copyright (c) 2024 Hemi Labs, Inc.
+//
+// This file is part of the posixutils-rs project covered under
+// the MIT License.  For the full license text, please see the LICENSE
+// file in the root directory of this project.
+// SPDX-License-Identifier: MIT
+//
+
+extern crate clap;
+extern crate plib;
+
+use clap::Parser;
+use gettextrs::{bind_textdomain_codeset, textdomain};
+use plib::PROJECT_NAME;
+use regex::{Regex, RegexBuilder};
```

Comment:
Major revision needed:   I do not think the `regex` crate can give us POSIX compliance?   I think the libc functions `regcomp` and `regexec` must be used to meet `BRE` vs `ERE` support requirements.